### ISSUE TITLE
First request always fail

### ISSUE THREAD

**Issue by mderazon** [loadimpact/k6#353]
``` shell
$  k6 run script.js --vus 1 --duration 10s
```
The script is doing a POST request, nothing fancy. 

The error I am getting is
```
WARN[0001] Request Failed                                error="Post https://example.com/users: stream error: stream ID 1; REFUSED_STREAM"
```

Running with Go debug flag:
```
$ GODEBUG=http2debug=2 k6 run script.js --vus 1
```

I am getting 
```
2017/10/29 14:39:30 http2: Transport failed to get client conn for example.com:443: http2: no cached connection was available
2017/10/29 14:39:31 http2: Transport creating client conn 0xc42227bba0 to 13.85.81.70:443
2017/10/29 14:39:31 http2: Framer 0xc42139b960: wrote SETTINGS len=18, settings: ENABLE_PUSH=0, INITIAL_WINDOW_SIZE=4194304, MAX_HEADER_LIST_SIZE=10485760
2017/10/29 14:39:31 http2: Framer 0xc42139b960: wrote WINDOW_UPDATE len=4 (conn) incr=1073741824
2017/10/29 14:39:31 http2: Transport encoding header ":authority" = "example.com"
2017/10/29 14:39:31 http2: Transport encoding header ":method" = "POST"
2017/10/29 14:39:31 http2: Transport encoding header ":path" = "/api"
2017/10/29 14:39:31 http2: Transport encoding header ":scheme" = "https"
2017/10/29 14:39:31 http2: Transport encoding header "accept" = "application/json, text/plain, */*"
2017/10/29 14:39:31 http2: Transport encoding header "content-type" = "application/json"
2017/10/29 14:39:31 http2: Transport encoding header "referer" = "http://other.com/app"
2017/10/29 14:39:31 http2: Transport encoding header "origin" = "http://other.com"
2017/10/29 14:39:31 http2: Transport encoding header "content-length" = "19"
2017/10/29 14:39:31 http2: Transport encoding header "accept-encoding" = "gzip"
2017/10/29 14:39:31 http2: Transport encoding header "user-agent" = "Go-http-client/2.0"
2017/10/29 14:39:31 http2: Framer 0xc42139b960: wrote HEADERS flags=END_HEADERS stream=1 len=162
2017/10/29 14:39:31 http2: Framer 0xc42139b960: wrote DATA stream=1 len=19 data="{\"linkName\":\"dana\"}"
2017/10/29 14:39:31 http2: Framer 0xc42139b960: wrote DATA flags=END_STREAM stream=1 len=0 data=""
2017/10/29 14:39:31 http2: Framer 0xc42139b960: read SETTINGS len=18, settings: MAX_CONCURRENT_STREAMS=128, INITIAL_WINDOW_SIZE=0, MAX_FRAME_SIZE=16777215
2017/10/29 14:39:31 http2: Transport received SETTINGS len=18, settings: MAX_CONCURRENT_STREAMS=128, INITIAL_WINDOW_SIZE=0, MAX_FRAME_SIZE=16777215
2017/10/29 14:39:31 http2: Framer 0xc42139b960: wrote SETTINGS flags=ACK len=0
2017/10/29 14:39:31 http2: Framer 0xc42139b960: read WINDOW_UPDATE len=4 (conn) incr=2147418112
2017/10/29 14:39:31 http2: Transport received WINDOW_UPDATE len=4 (conn) incr=2147418112
2017/10/29 14:39:31 http2: Framer 0xc42139b960: read SETTINGS flags=ACK len=0900ms / 0s
2017/10/29 14:39:31 http2: Transport received SETTINGS flags=ACK len=0
2017/10/29 14:39:31 http2: Framer 0xc42139b960: read RST_STREAM stream=1 len=4 ErrCode=REFUSED_STREAM
2017/10/29 14:39:31 http2: Transport received RST_STREAM stream=1 len=4 ErrCode=REFUSED_STREAM
2017/10/29 14:39:31 RoundTrip failure: stream error: stream ID 1; REFUSED_STREAM
```

See the last lines:
> 2017/10/29 14:39:31 RoundTrip failure: stream error: stream ID 1; REFUSED_STREAM

Looking it up leads me to this: https://github.com/golang/go/issues/17066
(I am running the test agains a Nginx server)
The rabbit hole continues to here: https://trac.nginx.org/nginx/ticket/959

I am not sure what this problem is about, but I did notice something.

**It's always the first batch of requests that are being rejected with this error**. So if for example I am doing 

```
$  k6 run script.js --vus 1 --duration 10s
```

I will get 1 error. But if I am doing

```
$  k6 run script.js --vus 3 --duration 10s
```

I will get 3 errors (the first 3 requests)
Problem is that when running hundreds or req/sec the warnings pollute the console very much.

Does anyone know what's the deal here? Do I need to make an OPTIONS request before each test ?

Thanks.

**Comment by liclac** [comment]
Sorry about the late response, I'm very confused by what's going on here, but I don't have time to look into it right now…

**Comment by mderazon** [comment]
tl;dr when running load on Nginx I am getting warning from the underlying go http lib. I get 1 line of warning for each request batch so the the screen fills with warnings pretty quickly.

**Comment by liclac** [comment]
Closing this until further notice, feel free to reopen if it's still happening.
